src/main/scala/no/ndla/imageapi/service/ImageConverter.scala

```diff
@@ -82,19 +82,20 @@ trait ImageConverter {
       (max(start - endRem, 0), min(end + startRem, originalDimensionSize))
     }
 
-    def dynamicCrop(image: ImageStream, percentFocalPoint: PercentPoint, targetWidthOpt: Option[Int], targetHeightOpt: Option[Int]): Try[ImageStream] = {
+    def dynamicCrop(image: ImageStream, percentFocalPoint: PercentPoint, targetWidthOpt: Option[Int], targetHeightOpt: Option[Int], ratioOpt: Option[Double]): Try[ImageStream] = {
       val sourceImage = image.sourceImage
       val focalPoint = toPixelPoint(percentFocalPoint, sourceImage)
       val (imageWidth, imageHeight) = (sourceImage.getWidth, sourceImage.getHeight)
 
-      val (targetWidth: Int, targetHeight: Int) = (targetWidthOpt, targetHeightOpt) match {
-        case (None, None) => return Success(image)
-        case (Some(w), Some(h)) => (min(w, imageWidth), min(h, imageHeight))
-        case (Some(w), None) =>
+      val (targetWidth: Int, targetHeight: Int) = (targetWidthOpt, targetHeightOpt, ratioOpt) match {
+        case (_, _, Some(ratio)) => minimalCropSizesToPreserveRatio(imageWidth, imageHeight, ratio)
```

Comment:
Hmm...usikker her, men når man sier `case (_, _, Some(ratio)) => ` så betyr det whatever i de to første, men en `Some()` i siste..?

Når vi lengre nede i matchingen sier `case (Some, Some, _) => ` ...hvilken av de to vinner? Den første?

Reply:
Riktig. Den første vinner hvis `ratioOpt` er definert. Da bryr vi oss ikke om resten. Og i alle de andre tilfellene ignorerer vi hva `ratioOpt` måtte være.